controllers/flinkcluster_util.go

```diff
@@ -246,3 +367,87 @@ func isUserControlFinished(controlStatus *v1beta1.FlinkClusterControlStatus) boo
 	return controlStatus.State == v1beta1.ControlStateSucceeded ||
 		controlStatus.State == v1beta1.ControlStateFailed
 }
+
+// Check if the savepoint has been created recently.
+func isSavepointUpToDate(now time.Time, jobStatus v1beta1.JobStatus) bool {
+	if jobStatus.SavepointLocation != "" && jobStatus.LastSavepointTime != "" {
+		tc := &TimeConverter{}
+		completedTime := tc.FromString(jobStatus.LastSavepointTime)
+		availableTime := completedTime.Add(time.Duration(int64(SavepointAgeForJobUpdateSec) * int64(time.Second)))
+		if now.Before(availableTime) {
+			return true
+		}
+	}
+	return false
+}
+
+func isComponentUpdated(component runtime.Object, cluster v1beta1.FlinkCluster) bool {
```

Comment:
Add comment for this function, in particular how it measures if a component has been updated.

Reply:
Done.

Reply:
Avoid using `o` which is not easy to be distinguished with `0`, s/o/obj?